flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/CheckpointStatistics.java

```diff
@@ -0,0 +1,763 @@
+		private final long failureTimestamp;
+
+		@JsonProperty(FIELD_NAME_FAILURE_MESSAGE)
+		@Nullable
+		private final String failureMessage;
+
+		@JsonCreator
+		public FailedCheckpointStatistics(
+				@JsonProperty(FIELD_NAME_ID) long id,
+				@JsonProperty(FIELD_NAME_STATUS) CheckpointStatsStatus status,
+				@JsonProperty(FIELD_NAME_IS_SAVEPOINT) boolean savepoint,
+				@JsonProperty(FIELD_NAME_TRIGGER_TIMESTAMP) long triggerTimestamp,
+				@JsonProperty(FIELD_NAME_LATEST_ACK_TIMESTAMP) long latestAckTimestamp,
+				@JsonProperty(FIELD_NAME_STATE_SIZE) long stateSize,
+				@JsonProperty(FIELD_NAME_DURATION) long duration,
+				@JsonProperty(FIELD_NAME_ALIGNMENT_BUFFERED) long alignmentBuffered,
+				@JsonProperty(FIELD_NAME_NUM_SUBTASKS) int numSubtasks,
+				@JsonProperty(FIELD_NAME_NUM_ACK_SUBTASKS) int numAckSubtasks,
+				@JsonProperty(FIELD_NAME_FAILURE_TIMESTAMP) long failureTimestamp,
+				@JsonProperty(FIELD_NAME_FAILURE_MESSAGE) @Nullable String failureMessage) {
+			super(id, status, savepoint, triggerTimestamp, latestAckTimestamp, stateSize, duration, alignmentBuffered, numSubtasks, numAckSubtasks);
+
+			this.failureTimestamp = failureTimestamp;
+			this.failureMessage = failureMessage;
+		}
+
+		public long getFailureTimestamp() {
+			return failureTimestamp;
+		}
+
+		@Nullable
+		public String getFailureMessage() {
+			return failureMessage;
+		}
+
+		@Override
+		public boolean equals(Object o) {
+			if (this == o) {
+				return true;
+			}
+			if (o == null || getClass() != o.getClass()) {
+				return false;
+			}
+			if (!super.equals(o)) {
+				return false;
+			}
+			FailedCheckpointStatistics that = (FailedCheckpointStatistics) o;
+			return failureTimestamp == that.failureTimestamp &&
+				Objects.equals(failureMessage, that.failureMessage);
+		}
+
+		@Override
+		public int hashCode() {
+			return Objects.hash(super.hashCode(), failureTimestamp, failureMessage);
+		}
+	}
+
+	/**
+	 * Statistics for a restored checkpoint.
+	 */
+	public static final class RestoredCheckpointStatistics {
+
+		public static final String FIELD_NAME_ID = "id";
+
+		public static final String FIELD_NAME_RESTORE_TIMESTAMP = "restore_timestamp";
+
+		public static final String FIELD_NAME_IS_SAVEPOINT = "is_savepoint";
+
+		public static final String FIELD_NAME_EXTERNAL_PATH = "external_path";
+
+		@JsonProperty(FIELD_NAME_ID)
+		private final long id;
+
+		@JsonProperty(FIELD_NAME_RESTORE_TIMESTAMP)
+		private final long restoreTimestamp;
+
+		@JsonProperty(FIELD_NAME_IS_SAVEPOINT)
+		private final boolean savepoint;
+
+		@JsonProperty(FIELD_NAME_EXTERNAL_PATH)
+		@Nullable
+		private final String externalPath;
+
+		@JsonCreator
+		public RestoredCheckpointStatistics(
+			@JsonProperty(FIELD_NAME_ID) long id,
+			@JsonProperty(FIELD_NAME_RESTORE_TIMESTAMP) long restoreTimestamp,
+			@JsonProperty(FIELD_NAME_IS_SAVEPOINT) boolean savepoint,
+			@JsonProperty(FIELD_NAME_EXTERNAL_PATH) @Nullable String externalPath) {
```

Comment:
indentation inconsistent with other constructors

Reply:
Thanks for catching this. Will fix it.